### DISCUSSION TITLE
task runs even not all upstream has done in trigger rule ALL_SUCCESS

### DISCUSSION THREAD

**Discussion by robinhur** [apache/airflow#21921] [Q&A]
This DAG is already have been using for quite a long time, and all were working normally.

However, after modifying some codes(task structures), the following abnormal symptoms occurred.

- One task has 14 upstream tasks and has 'ALL_SUCCESS' trigger rule that is executed after all of them are finished. (worked well before) 
- But after we modified the DAG, the task will be executed even though some of the 14 tasks have failed.
- In TASK INSTANCE DETAILS menu, it is indicated that the conditions are not satisfied because some tasks failed, but in the TASK LOG, it says all dependencies are met for the task.

**Task Instance Details**
```
Task Instance State	
Task is in the 'success' state which is not a valid state for execution. 
The task must be cleared in order to be run.

Trigger Rule
Task's trigger rule 'all_success' requires all upstream tasks to have succeeded, 
but found 3 non-success(es). 
upstream_tasks_state=
{'total': 14, 
'successes': Decimal('11'), 
'skipped': Decimal('0'), 
'failed': Decimal('3'), 
'upstream_failed': Decimal('0'), 
'done': 14}, 
upstream_task_ids=...
```

**Task Log**
```
...
[2022-02-24 .. {taskinstance.py:670} INFO - Dependencies all met for <TaskInstance: ...
...
```

Which part should I take a look?

**Comment by potiuk** [comment]
This is a very vague description. I think you mixed up some diferent tasks/schedules there. Hard to say what the problem might be as your code is very selective. Very hard to help based on that.
